File: Python3/Easy/RuntimeError/x-of-a-kind-in-a-deck-of-cards_1.py
class Solution:
    def hasGroupsSizeX(self, deck: List[int]) -> bool:
        
        # deck = [1,2,3,4,4,3,2,1]
        # deck = [1,1,1,2,2,2,3,3]
        # deck = [1]
        # deck = [1,1]
        # deck = [1,1,2,2,2,2]
        # deck = [1,1,1,1,2,2,2,2,2,2]
        # deck = [1,1,1,1,2,2,2,2,2,2,3,3,3,3,3,3,3,3,3,3]
        # deck = [1,1,1,1,2,2,2,2,2,2,3,3,3,3,3,3,3,3,3,3,4,4,4,4,4,4,4,4,4,4,4,4]
        # deck = [1,1,1,1,2,2,2,2,2,2,3,3,3,3,3,3,3,3,

File: Rust/Easy/WrongAnswer/x-of-a-kind-in-a-deck-of-cards_1.rs
impl Solution {
    pub fn has_groups_size_x(deck: Vec<i32>) -> bool {
        
        use std::collections::HashMap;
        let mut map = HashMap::new();
        for i in deck {
            *map.entry(i).or_insert(0) += 1;
        }
        let mut v = map.values().collect::<Vec<&i32>>();
        v.sort();
        let mut i = 0;
        let mut j = 1;
        while j < v.len() {
            if v[i] != v[j] {
                return false;
            }
            i += 1;
            j += 1;
        }
        true
    }
}